internal/server/polling/terraform.go

```diff
@@ -91,17 +91,19 @@ func (s *Server) reconcileTerraform(ctx context.Context, originalTF *infrav1.Ter
 			Namespace: originalTF.Namespace,
 		},
 	}
-	branchLabels := s.createLabels(originalTF.Labels, originalTF.Name, branch, prID)
-	branchSecretName := s.createObjectName(originalTF.Spec.WriteOutputsToSecret.Name, branch, prID)
 
+	branchLabels := s.createLabels(originalTF.Labels, originalTF.Name, branch, prID)
 	op, err := controllerutil.CreateOrUpdate(ctx, s.clusterClient, tf, func() error {
 		spec := originalTF.Spec.DeepCopy()
 
 		spec.SourceRef.Name = source.Name
 		spec.SourceRef.Namespace = source.Namespace
 		spec.PlanOnly = true
 		spec.StoreReadablePlan = "human"
-		spec.WriteOutputsToSecret.Name = branchSecretName
+		// relocate the output secret, so it's not shared between branches
+		if spec.WriteOutputsToSecret != nil && originalTF.Spec.WriteOutputsToSecret != nil {
```

Comment:
Do we need to handle the case where `originalTF.Spec.WriteOutputsToSecret` is nil?

Reply:
It's a safe guard, so yes

Reply:
Ok are we going to include this case (`originalTF.Spec.WriteOutputsToSecret` being nil) as part of this PR?

Reply:
I think I mistook your question.

In case of `originalTF.Spec.WriteOutputsToSecret` is nil it means that no configuration is required. That's why we don't have the else case here.